supervisor/supervisor.go

```diff
@@ -0,0 +1,41 @@
+package supervisor
+
+import (
+	"context"
+	"github.com/open-telemetry/opamp-go/supervisor/types"
+)
+
+// Supervisor allows the management of an agent. Specific agents like an
+// OpenTelemetry collector can implement a custom supervisor that understands
+// the internal workings of the collector. Alternatively, a default supervisor
+// allows for the supervision of arbitrary daemon-like processes and provides
+// management options like remote configuration change, agent upgrades,
+// watchdog capabilities, telemetry reporting, etc.
+//
+// The supervisor wraps the client part of the OpAMP protocol and therefore
+// requires an OpAMP server to be running at some endpoint. There is a 1:1
+// mapping between a supervisor and the OpAMP client. A single supervisor
+// executable, however, can run several supervisors within it thereby managing
+// several distinct agent types.
```

Comment:
I am not sure this is necessary. What's the use case for multiple Agents per Supervisor executable?